Exercises/Chapter-4.txt

```diff
@@ -101,4 +101,70 @@ a = ('a'..'z').to_a
  => ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"] 
 2.2.3 :045 > a[7]
  => "h" 
-
+ 
+ 
+ 
+ 
+ 
+1. Using the range 0..16, print out the first 17 powers of 2.
+2.2.3 :021 > (0..16).each do |i|
+2.2.3 :022 >     puts i**2
+2.2.3 :023?>   end
+0
+1
+4
+9
+16
+25
+36
+49
+64
+81
+100
+121
+144
+169
+196
+225
+256
+ => 0..16 
+
+
+2. Define a method called yeller that takes in an array of characters and returns a string with an ALLCAPS version of the input. Verify that yeller([’o’, ’l’, ’d’]) returns "OLD". Hint: Combine map, upcase, and join.
+2.2.3 :001 > def yeller(s)
+2.2.3 :002?>   s.map(&:upcase).join
+2.2.3 :003?>   end
+ => :yeller 
+2.2.3 :004 > yeller(['o','l','d'])
+ => "OLD" 
+
+NOTE TO SPINOSA: I have no idea why this worked and the below method didn't.
```

Comment:
we covered this IRL

@joeyevoli TODO: explain what's wrong / what's going on / what's different in the below method